### PR TITLE
feat: adding keywords page

### LINKED ISSUE #815: ✨ Enhancement: It well be great to provide a keywords page
### Is your feature request related to a problem? Please describe

We don't have a list of keywords in a single page and users will benefit from it.

### Describe the solution you'd like

Taking advantage of the work done by the GSoD team reviewing the keywords, I'd like to provide a webpage with a list of all keywords and the links of pages referencing them. The list is available in this doc.

[Identify Keywords - JSON Schema Reference.pdf](https://github.com/user-attachments/files/16335092/Identify.Keywords.-.JSON.Schema.Reference.pdf)

The page should be totally data driven so we'll create first a yaml file in the data folder each keyword and the links (text+url). Once we have the data we'll build the page with a structure like this:

Title: JSON Schema Keywords
Description: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat

(List of keywords sorted alphabetically)

* Keyword 1 (Link to the keyword in Learn JSON Schema)
  * [Link 1](#)
  * [Link N](#)
* Keyword 2
  * [Link 1](#)
  * [Link N](#)
* Keyword N
  * [Link 1](#)
  * [Link N](#)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
**What kind of change does this PR introduce?**

Adding new keywords page

**Issue Number:**

-  Closes #815

**Screenshots/videos:**
image

**If relevant, did you update the documentation?**
N/A

**Summary**

A new keywords page is added.

### CODE REVIEW

File: data/keywords.yml [hunk 1]
```diff
@@ -0,0 +1,248 @@
+- title: JSON Schema Keywords
```
Comment (benjagm) [inline data/keywords.yml:1]: We dont need title of the page and description here ... it does not make sense. This file is only for keywords.

File: data/keywords.yml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  description: >-
+    JSON Schema keywords are the building blocks of JSON Schema. They are used
+    to define the structure of a JSON document
+  categorys:
```
Comment (benjagm) [inline data/keywords.yml:5]: We need to change the structure to store URLs and titles of the URLs and also allow 1 keyword to belong to multiple vocabularies without having to duplicate it. Find below an example :

````
- name: $anchor
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/learn/json-schema-examples#ecommerce-system'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/structuring#anchor'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/anchor'
      title:
- name: $comment
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/understanding-json-schema/reference/comments#comments'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/comment'
      title:
- name: $defs
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/learn/miscellaneous-examples'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/structuring#defs'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/defs'
      title:
- name: $dynamicAnchor
  vocabulary: [core]
  links:
    - url: 'https://www.learnjsonschema.com/2020-12/core/dynamicanchor'
      title:
- name: $dynamicRef
  vocabulary: [core]
  links:
    - url: 'https://www.learnjsonschema.com/2020-12/core/dynamicref'
      title:
- name: $id
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/learn/getting-started-step-by-step#create-a-schema-definition'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/basics#declaring-a-unique-identifier'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/id'
      title:
- name: $ref
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/learn/getting-started-step-by-step#create-a-schema-definition'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/basics#declaring-a-json-schema'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/reference/array#unevaluateditems'
      title:
    - url: 'https://json-schema.org/understanding-json-schema/structuring#id'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/ref'
      title:
- name: $schema
  vocabulary: [core]
  links:
    - url: 'https://json-schema.org/understanding-json-schema/reference/schema'
      title:
    - url: 'https://www.learnjsonschema.com/2020-12/core/schema'
      title:
````

File: pages/keywords/index.page.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
```
Comment (benjagm) [inline pages/keywords/index.page.tsx:1]: Some comments:

- keywords should be shorted alphabetically without any grouping (vocabulary is only a property)
- Avoid using a table. I provided an example using an unsorted list (bullet points). Plese check the issue
- For the links we should store links and titles. Please show the titles and create links with the URLs to provide a most friendly approach for showing urls.

**Review by benjagm — CHANGES_REQUESTED** [review]
Great first version! Congrats. I left some comments to make some adjustments but we are very close!

**Review by benjagm — APPROVED** [review]
Looks great! Amazing job @DhairyaMajmudar !

**Review by jdesrosiers — APPROVED** [review]
This looks like a really useful resource!

I do worry about it getting out of date as the website evolves. Is there a checklist for adding things the to website? Having something that reminds to check if this index needs to be updated would be a good idea.

**Pull request merged**